File: api/kv-quiz/src/main/java/in/kvapps/kv_quiz/dto/UserProgressDto.java
package in.kvapps.kv_quiz.dto;

import lombok.Data;

import java.util.Objects;

@Data
public class UserProgressDto {
    private String name;
    private int progress;
    private boolean testCompleted;
    private int answeredQuestions;
    private int correctAnswers;
    private int totalQuestions;

    @Override
    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (!(obj instanceof UserProgressDto)) return false;
        UserProgressDto user = (UserProgressDto) obj;
        return Objects.equals(name, user.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name);
    }
}

File: ui/express-server/kv-quiz-app/src/app/dto/UserProgressDto.ts
export class UserProgressDto {
  name: string = '';
  progress: number = 0;
  testCompleted: boolean = false;
  answeredQuestions: number = 0;
  correctAnswers: number = 0;
  totalQuestions: number = 0;

  constructor(name: string, progress: number, testCompleted: boolean, answeredQuestions: number, correctAnswers: number, totalQuestions: number) {
    this.name = name;
    this.progress = progress;
    this.testCompleted = testCompleted;
    this.answeredQuestions = answeredQuestions;
    this.correctAnswers = correctAnswers;
    this.totalQuestions = totalQuestions;
  }  
}